### PR TITLE
doc: convert weechat docs to commonmark

### LINKED ISSUE #105264: Convert Weechat from Docbook to CommonMark
This issue tracks the conversion of `/doc/builders/packages/weechat.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/builders/packages/index.xml`: change `weechat.xml` to `weechat.section.xml`.
4. Rename `doc/builders/packages/weechat.xml` to `weechat.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by fzakaria**
###### Motivation for this change

This is essentially an automatic pandoc conversion of the weechat
section in docbook as part of the larger [CommonMark Docs project](https://github.com/NixOS/nixpkgs/projects/37)

I generated the resulting documentation and verified it looks correct to what was previously present at https://nixos.org/manual/nixpkgs/unstable/#sec-weechat 

I also followed https://github.com/NixOS/nixpkgs/issues/105243 by running _prettier_ on the commonmark generated.

image

fixes #105264

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@ryantm

### CODE REVIEW

**Comment by fzakaria** [comment]
@ryantm i'm curious why the CSS layout of what I generate looks so different than the nixos.org styles?
It was much more narrow

image

File: doc/builders/packages/weechat.section.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+```
+
+If the `configure` function returns an attrset without the `plugins` attribute, `availablePlugins` will be used automatically.
+
+The plugins currently available are `python`, `perl`, `ruby`, `guile`, `tcl` and `lua`.
+
+The python and perl plugins allows the addition of extra libraries. For instance, the `inotify.py` script in weechat-scripts requires D-Bus or libnotify, and the `fish.py` script requires pycrypto. To use these scripts, use the plugin's `withPackages` attribute:
+
+```nix
```
Comment (fzakaria) [inline doc/builders/packages/weechat.section.md:18]: Looks like _pandoc_ choses to use indentation for codeblocks and doesn't have then the _nix_ language fence block.
I updated the PR and fixed up all the code blocks.

**Comment by jonringer** [comment]
> i'm curious why the CSS layout of what I generate looks so different than the nixos.org styles

nixos.org uses it's own style sheets

File: doc/builders/packages/weechat.section.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    plugins = with availablePlugins; [ python perl ];
+  }
+}
+```
+
+If the `configure` function returns an attrset without the `plugins` attribute, `availablePlugins` will be used automatically.
+
+The plugins currently available are `python`, `perl`, `ruby`, `guile`, `tcl` and `lua`.
+
+The python and perl plugins allows the addition of extra libraries. For instance, the `inotify.py` script in weechat-scripts requires D-Bus or libnotify, and the `fish.py` script requires pycrypto. To use these scripts, use the plugin's `withPackages` attribute:
```
Comment (FRidh) [inline doc/builders/packages/weechat.section.md:16]:
```suggestion
The python and perl plugins allows the addition of extra libraries. For instance, the `inotify.py` script in `weechat-scripts` requires D-Bus or libnotify, and the `fish.py` script requires `pycrypto`. To use these scripts, use the plugin's `withPackages` attribute:
```

File: doc/builders/packages/weechat.section.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+# Weechat
```
Comment (jtojnar) [inline doc/builders/packages/weechat.section.md:1]: Missing anchor:

```suggestion
# Weechat {#sec-weechat}
```

**Review by ryantm — APPROVED** [review]
Source and rendered looks good!

**Pull request merged**